scripts/dev/generate_config_types.py

```diff
@@ -0,0 +1,237 @@
+        "from __future__ import annotations",
+        "import re",
+        "from dataclasses import dataclass",
+        "from collections.abc import Mapping",
+        "from typing import Optional, Union, Literal",
+        "",
+        "",
+    ]
+
+    def generate_class(
+        class_name: str,
+        config: NestedConfig,
+        *,
+        indent: str = "",
+        description: str | None = None,
+    ) -> list[str]:
+        class_def = [
+            f"{indent}@dataclass",
+            f"{indent}class {class_name}:",
+        ]
+        if description is not None:
+            class_def.append(f"{indent}    {triple_quote(description)}")
+
+        for key, value in config.items():
+            if isinstance(value, Option):
+                type_hint = get_type_hint(value.typ)
+                if value.default is None:
+                    class_def.append(f"{indent}    {key}: Optional[{type_hint}]")
+                else:
+                    class_def.append(f"{indent}    {key}: {type_hint}")
+
+                if value.description:
+                    lines = value.description.split("\n")
+                    description = "\n\n".join(
+                        [
+                            line if i == 0 else f"{indent}    {line}"
+                            for i, line in enumerate(lines)
+                        ]
+                    )
+                    if not description.endswith("\n") and len(lines) > 1:
+                        description += f"\n{indent}    "
+
+                    class_def.append(f"{indent}    {triple_quote(description)}\n")
+            elif isinstance(value, dict):
+                nested_class_name = "_" + snake_to_camel(key)
+                class_def.append(f"{indent}    {key}: {nested_class_name}")
+                class_def.extend(
+                    generate_class(nested_class_name, value, indent=indent + "    ")
+                )
+            else:
+                assert_never(value)
+
+        return class_def
+
+    nested_config = make_nested_config(config_data)
+    output.extend(
+        generate_class(
+            "ConfigContainer",
+            nested_config,
+            description="Type for the `c` variable in `config.py`.",
+        )
+    )
+
+    return "\n".join(output)
+
+
+def snake_to_camel(name: str) -> str:
+    return "".join(word.capitalize() for word in name.split("_"))
+
+
+def literal(v: str) -> str:
+    return '"' + v + '"'
+
+
+def triple_quote(v: str) -> str:
+    r"""surround the given string with trible double quotes."""
+    # some option descriptions use `\+` which isn't a valid escape sequence
+    # in python docstrings, so just escape it.
+    return '"""' + v.replace("\\+", "\\\\+") + '"""'
+
+
+def main():
+    configdata.init()
+
+    generated_code = generate_config_types(configdata.DATA)
+
+    output_file = (
+        Path(__file__).parent.parent.parent
+        / "qutebrowser"
+        / "config"
+        / "configcontainer.py"
```

Comment:
```suggestion
        / "configcontainer_types.py"
```

or so maybe, to make it clearer from the filename that this isn't any actual implementation?